compiler/optimizer/LocalOpts.cpp

```diff
@@ -3386,6 +3386,49 @@ int32_t TR_EliminateRedundantGotos::process(TR::TreeTop *startTree, TR::TreeTop
          //////dumpOptDetails(comp(), "Contains trees other than gotos when at block number : %d\n", block->getNumber());
          }
 
+      bool blockIsRegionEntry = false;
+      TR_Structure *rootStructure = cfg->getStructure();
+      TR_RegionStructure *blockRegion = NULL;
+      if (rootStructure != NULL)
+         {
+         blockRegion = block->getParentStructureIfExists(cfg);
+         blockIsRegionEntry =
+            blockRegion != NULL && blockRegion->getNumber() == block->getNumber();
+         }
+
+      bool manuallyFixStructure = blockIsRegionEntry;
+
+      if (block->isExtensionOfPreviousBlock())
```

Comment:
Could you please elaborate on why we set the bool to manually fix structure to false in this case (I understand that the test guarantees that it cannot be a loop header but am missing the connection to why that matters wrt manually fixin structure).

Reply:
The intention was to avoid unnecessarily failing [this test in the next commit](https://github.com/eclipse/omr/pull/2444/commits/9be5d861d71e74ca3f8113aa3455f1d342213a62#diff-bf9bdb533ff9e25408c9fac330ac7aedR3495), which says that when manually fixing structure, we have to be able to adjust all predecessors.

But come to think of it again, it seems it doesn't help to change `manuallyFixStructure` here. If the block is an extension, it has exactly one predecessor and that predecessor falls through. We only accept fallthroughs for empty blocks, so the block would fail that test because of `emptyBlock` anyway.

Reply:
Updated to remove the unnecessary change to `manuallyFixStructure` in this case